File: Java/src/com/dbc/algorithm/Daily/CanPlaceFlowers.java
package com.dbc.algorithm.Daily;

public class CanPlaceFlowers {
    public static boolean canPlaceFlowers(int[] flowerbed, int n) {
        int count = 0, length = flowerbed.length, prev = -1;
        for (int i = 0; i < length; i++) {
            if (flowerbed[i] == 1) {
                if (prev < 0) {
                    count += i / 2;
                } else {
                    count += (i - prev - 2) / 2;
                }
                if (count >= n)
                    return true;
                prev = i;
            }
        }
        if (prev < 0) {
            count += (length - prev) / 2;
        } else {
            count += (length - prev - 1) / 2;
        }
        return count >= n;
    }
}

File: Python/can_place_flowers.py
def canPlaceFlowers(flowerbed: list, n: int) -> bool:
    count, m, prev = 0, len(flowerbed), -1
    for i in range(m):
        if flowerbed[i] == 1:
            if prev < 0:
                count += i // 2
            else:
                count += (i - prev - 2) // 2
            if count >= n:
                return True
            prev = i
    if prev < 0:
        count += (m + 1) // 2
    else:
        count += (m - prev - 1) // 2

    return count >= n


if __name__ == "__main__":
    print(canPlaceFlowers(flowerbed=[1, 0, 0, 0, 1], n=2))
